lib/bucket_store/in_memory.rb

```diff
@@ -24,21 +24,20 @@ def reset!
       @buckets = Hash.new { |hash, key| hash[key] = {} }
     end
 
-    def upload!(bucket:, key:, content:)
-      @buckets[bucket][key] = content
+    def upload!(bucket:, key:, file:)
+      @buckets[bucket][key] = file.read
 
       {
         bucket: bucket,
         key: key,
       }
     end
 
-    def download(bucket:, key:)
-      {
-        bucket: bucket,
-        key: key,
-        content: @buckets[bucket].fetch(key),
-      }
+    def download(bucket:, key:, file:)
+      file.tap do |f|
+        f.write(@buckets[bucket].fetch(key))
+        f.rewind
```

Comment:
We need to do something similar in the disk adapter as well. I was also testing GCS locally and I get inconsistent behaviour now via the emulator (as in - sometimes I get the right output, sometimes don't). Might be worth popping a `.rewind` in the gcs adapter too while we're here to make sure we always get a readable stream back.

Reply:
✔️